pkg/distributor/ha_tracker.go

```diff
@@ -471,7 +471,9 @@ func findHALabels(replicaLabel, clusterLabel string, labels []cortexpb.LabelAdap
 			replica = pair.Value
 		}
 		if pair.Name == clusterLabel {
-			cluster = pair.Value
+			// Cloning ensures that the cluster string can be exposed in metrics
+			// which has a longer life time than a single request.
```

Comment:
I think we should just mention yoloString, additionally long life time of string is not really a  problem per-se, but long life lived string that retain lots garbage is not. I would recommend the comment to be something like "cluster label is unmarshalled into yoloString, which retains original remote write request body in memory. Hence, we clone the yoloString to allow  the request body to be GC'ed". The key thing is to mention 

1. yoloString
1. requestbody retention
1. why clone